app/src/main/java/org/simple/clinic/pin/BruteForceProtection.kt

```diff
@@ -0,0 +1,92 @@
+package org.simple.clinic.pin
+
+import com.f2prateek.rx.preferences2.Preference
+import io.reactivex.Completable
+import io.reactivex.Observable
+import io.reactivex.Single
+import io.reactivex.rxkotlin.Observables
+import org.simple.clinic.pin.BruteForceProtection.ProtectedState.Allowed
+import org.simple.clinic.pin.BruteForceProtection.ProtectedState.Blocked
+import org.simple.clinic.util.Just
+import org.simple.clinic.util.None
+import org.simple.clinic.util.Optional
+import org.threeten.bp.Clock
+import org.threeten.bp.Instant
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+import javax.inject.Named
+
+class BruteForceProtection @Inject constructor(
+    private val clock: Clock,
+    private val configProvider: Single<BruteForceProtectionConfig>,
+    @Named("pin_failed_auth_count") private val failedAuthCountPreference: Preference<Int>,    // TODO:
```

Comment:
> I think we can drop the Preference since it's already encoded in the type and the names are more explicit about what they represent.

True

> what do you feel about `numberOfTimesIncorrectPinEntered` and `incorrectPinEntryLimitReachedAt`?

No… that looks too long to me. How about `failedAttemptsCount`?

Reply:
The problem with `failedAttemptsCount` is that when I look at it, I don't know what the "attempt" is for.

Reply:
Does `BruteForceProtection` class even need to know what is it protecting from brute force?

Reply:
Even if it does, it is not obvious that it's being used with PIN entry? 

We could use `failedAuthAttemptsCount`?

Reply:
@vinaysshenoy is it okay if I discuss/make these changes in my next PR? The feature is behind a flag anyway.

Reply:
Yup.